libs/features/richresults.js

```diff
@@ -1,26 +1,61 @@
-// createTag and getMetadata are passed in to avoid circular dependencies
-export function addRichResults(type, { createTag, getMetadata }) {
-  const addRichResultsForNewsArticle = () => {
-    const newsArticle = {
-      '@context': 'https://schema.org',
-      '@type': 'NewsArticle',
-      headLine: getMetadata('og:title'),
-      image: getMetadata('og:image'),
-      datePublished: getMetadata('published'),
-      dateModified: getMetadata('modified'),
-      author: {
-        '@type': 'Person',
-        name: getMetadata('authorname'),
-        url: getMetadata('authorurl'),
+function getRichResultsForNewsArticle(getMetadata) {
+  return {
+    '@context': 'https://schema.org',
+    '@type': 'NewsArticle',
+    headLine: getMetadata('og:title'),
+    image: getMetadata('og:image'),
+    datePublished: getMetadata('published'),
+    dateModified: getMetadata('modified'),
+    author: {
+      '@type': 'Person',
+      name: getMetadata('authorname'),
+      url: getMetadata('authorurl'),
+    },
+  };
+}
+
+function getRichResultsForSiteSearchBox(getMetadata) {
+  // See specifications at https://developers.google.com/search/docs/appearance/structured-data/sitelinks-searchbox
+  const SEARCH_TERM_STRING = 'search_term_string';
+  const urlTemplate = `${getMetadata('search-url')}?${getMetadata('search-parameter-name')}={${SEARCH_TERM_STRING}}`;
+  return {
+    '@context': 'https://schema.org',
+    '@type': 'WebSite',
+    url: getMetadata('url'),
+    potentialAction: {
+      '@type': 'SearchAction',
+      target: {
+        '@type': 'EntryPoint',
+        urlTemplate: urlTemplate,
```

Comment:
nit: you could just write this in object-shorthand form 
```
'@type': 'EntryPoint',
urlTemplate,
```

Reply:
Addressed with https://github.com/adobecom/milo/pull/389/commits/4202377535cd937ec2c29bf8719c23dc4abda6e1